### PR TITLE
Data anonymisation

### PR DESCRIPTION
**Pull request by ssrihari**

### CODE REVIEW

File: lib/tasks/anonymize.rake [hunk 1]
```diff
@@ -39,37 +39,99 @@ namespace :anonymize do
...
         whitelist 'sync_approval_status'
         anonymize('full_name').using FieldStrategy::RandomFullName.new
+        anonymize('phone_number').using FieldStrategy::FormattedStringNumber.new
+        anonymize('password_digest').using FieldStrategy::RandomFormattedString.new
+        scramble('users', 'otp')
+        scramble('users', 'otp_valid_until')
+        scramble('users', 'access_token')
+        scramble('users', 'logged_in_at')
+        scramble('users', 'registration_facility_id')
+        scramble('users', 'sync_approval_status_reason')
```
Comment (divs1210) [inline lib/tasks/anonymize.rake]: whitelist

File: lib/tasks/anonymize.rake [hunk 2]
```diff
@@ -86,7 +148,16 @@ namespace :anonymize do
...
         primary_key 'id'
-        whitelist 'full_name', 'age', 'gender', 'status', 'address_id', 'date_of_birth'
+        scramble('patients', 'full_name')
+        scramble('patients', 'age')
+        scramble('patients', 'gender')
+        scramble('patients', 'status')
+        scramble('patients', 'address_id')
+        scramble('patients', 'date_of_birth')
+        scramble('patients', 'age_updated_at')
+        scramble('patients', 'registration_user_id')
```
Comment (divs1210) [inline lib/tasks/anonymize.rake]: whitelist

File: lib/tasks/anonymize.rake [hunk 3]
```diff
@@ -118,17 +189,35 @@ namespace :anonymize do
         primary_key 'id'
         whitelist 'facility_id'
         whitelist 'patient_id'
-      end
-
-      table 'user_facilities' do
-        primary_key 'id'
-        whitelist 'facility_id'
-        whitelist 'user_id'
+        scramble('appointments', 'cancel_reason')
```
Comment (divs1210) [inline lib/tasks/anonymize.rake]: club cancel_reason and status

File: lib/tasks/anonymize.rake [hunk 4]
```diff
@@ -86,7 +148,16 @@ namespace :anonymize do
 
       table 'patients' do
         primary_key 'id'
-        whitelist 'full_name', 'age', 'gender', 'status', 'address_id', 'date_of_birth'
+        scramble('patients', 'full_name')
+        scramble('patients', 'age')
```
Comment (divs1210) [inline lib/tasks/anonymize.rake]: club age, dob, age_updated

Comment (divs1210) [reply lib/tasks/anonymize.rake]: age / gender should be scramble?

File: lib/tasks/anonymize.rake [hunk 5]
```diff
@@ -39,37 +39,99 @@ namespace :anonymize do
...
         primary_key 'id'
         whitelist 'sync_approval_status'
         anonymize('full_name').using FieldStrategy::RandomFullName.new
+        anonymize('phone_number').using FieldStrategy::FormattedStringNumber.new
+        anonymize('password_digest').using FieldStrategy::RandomFormattedString.new
+        scramble('users', 'otp')
+        scramble('users', 'otp_valid_until')
+        scramble('users', 'access_token')
+        scramble('users', 'logged_in_at')
+        scramble('users', 'registration_facility_id')
```
Comment (divs1210) [inline lib/tasks/anonymize.rake]: whitelist

File: lib/tasks/anonymize.rake (hunk 1 above)
Comment (divs1210) [reply lib/tasks/anonymize.rake]: (sync_approval_status)

File: lib/tasks/anonymize.rake (hunk 4 above)
Comment (govindkrjoshi) [reply lib/tasks/anonymize.rake]: We should scramble age and gender since they can be used to identify patients along with BPs.

**Pull request merged**